src/devices/src/virtio/net/device.rs

```diff
@@ -409,24 +425,28 @@ impl Net {
     fn write_to_mmds_or_tap(
         mmds_ns: Option<&mut MmdsNetworkStack>,
         rate_limiter: &mut RateLimiter,
-        frame_buf: &[u8],
+        frame_buf: &mut [u8],
```

Comment:
I would strongly suggest to rename this for clarity. For example it misled and confused me for a while tying to figure out why this function was getting both `frame_buf` and `frame_iovec`.
```suggestion
        headers: &mut [u8],
```

Reply:
agreed